Library/Homebrew/cmd/prune.rb

```diff
@@ -47,5 +48,7 @@ def prune
       print "and #{d} directories " if d > 0
       puts "from #{HOMEBREW_PREFIX}"
     end unless ARGV.dry_run?
+
+    unlinkapps_prune(:dry_run => ARGV.dry_run?, :quiet => true)
```

Comment:
I'd prefer to do that in another PR as I expect some discussion. The reason is there are multiple places from where this likely needs to be called, though `uninstall` is the most obvious one.